boa/src/builtins/regexp/mod.rs

```diff
@@ -1125,19 +1218,24 @@ impl RegExp {
         let length_arg_str = arg_str.encode_utf16().count();
 
         // 5. Let functionalReplace be IsCallable(replaceValue).
-        let replace_value = args.get(1).cloned().unwrap_or_default();
+        let mut replace_value = args.get(1).cloned().unwrap_or_default();
         let functional_replace = replace_value.is_function();
 
         // 6. If functionalReplace is false, then
-        // a. Set replaceValue to ? ToString(replaceValue).
+        if !functional_replace {
+            // a. Set replaceValue to ? ToString(replaceValue).
+            replace_value = replace_value.to_string(context)?.into();
+        }
 
         // 7. Let global be ! ToBoolean(? Get(rx, "global")).
         let global = this.get_field("global", context)?.to_boolean();
 
         // 8. If global is true, then
-        // a. Let fullUnicode be ! ToBoolean(? Get(rx, "unicode")).
-        let unicode = this.get_field("unicode", context)?.to_boolean();
+        let mut unicode = false;
         if global {
+            // a. Let fullUnicode be ! ToBoolean(? Get(rx, "unicode")).
+            unicode = this.get_field("unicode", context)?.to_boolean();
+
```

Comment:
```suggestion
            // a. Let fullUnicode be ! ToBoolean(? Get(rx, "unicode")).
            unicode = rx.get("unicode", context)?.to_boolean();

```
We should call `GcObject::get` here. We can should extract the object

```rust
// 1. Let rx be the this value.
// 2. If Type(rx) is not Object, throw a TypeError exception.
let rx = if let Some(rx) = this.as_object() {
	rx
} else {
	return context.throw_type_error(
    	"RegExp.prototype[Symbol.replace] method called on incompatible value",
    );
};
```